metecho/api/models.py

```diff
@@ -554,8 +557,13 @@ def should_update_in_progress(self):
 
     def should_update_review(self):
         task_statuses = self.tasks.values_list("status", flat=True)
-        return task_statuses and all(
-            status == TASK_STATUSES.Completed for status in task_statuses
+        return (
+            task_statuses
+            and all(
+                status in [TASK_STATUSES.Completed, TASK_STATUSES.Cancelled]
+                for status in task_statuses
+            )
+            and any(status == TASK_STATUSES.Completed for status in task_statuses)
```

Comment:
@jerivas There's probably a more efficient way to do this. The goal is that this should return truthy if all tasks are completed or cancelled, but _not_ all tasks are cancelled.